### PR TITLE
Support more array methods

### PR DESCRIPTION
**Pull request by sklam**
Add:
* `array.ravel()`
* `array.flatten()`

Note: only support C layout for output for both methods

Depends on https://github.com/numba/numba/pull/1508

### CODE REVIEW

File: numba/tests/test_array_manipulation.py [hunk 1]
```diff
@@ -152,22 +159,32 @@ def test_flatten_array(self, flags=enable_pyobj_flags):
...
+            # compile
+            pyfunc = ravel_array
+            arraytype1 = typeof(a)
+            self.assertEqual(arraytype1.layout, assume_layout)
+            cr = compile_isolated(pyfunc, (arraytype1,), flags=flags)
+            cfunc = cr.entry_point
+
+            expected = pyfunc(a)
+            got = cfunc(a)
+            np.testing.assert_equal(expected, got)
```
Comment (pitrou) [inline numba/tests/test_array_manipulation.py]: You should also check whether a copy was made or not (depending on the layout).

File: numba/typing/arraydecl.py [hunk 2]
```diff
@@ -337,6 +337,20 @@ def resolve_view(self, ary, args, kws):
         retty = ary.copy(dtype=dtype)
         return signature(retty, *args)
 
+    @bound_function("array.ravel")
```
Comment (pitrou) [inline numba/typing/arraydecl.py]: What about the top-level function `np.ravel`?

**Comment by pitrou** [comment]
Thanks! This PR also needs a documentation update.

File: numba/typing/arraydecl.py (hunk 2 above)
Comment (sklam) [inline numba/typing/arraydecl.py]: Added in 7c87b37

File: numba/tests/test_array_manipulation.py (hunk 1 above)
Comment (sklam) [inline numba/tests/test_array_manipulation.py]: addressed in beacc87

**Pull request merged**